lib/js/jsconsole.nim

```diff
@@ -67,4 +69,15 @@ proc timeLog*(console: Console, label = "".cstring) {.importcpp.}
 proc table*(console: Console) {.importcpp, varargs.}
   ## https://developer.mozilla.org/docs/Web/API/Console/table
 
+since (1, 5):
+  # proc nor func can not be used here, because may eval assertion at compile-time,
+  # writing directly false or true to the JavaScript instead of the expression.
+  # importjs nor importcpp can not be used with template.
+
+  template doAssert*(console: Console; assertion) =
+    ## https://developer.mozilla.org/en-US/docs/Web/API/Console/assert
+    runnableExamples: console.doAssert(42 == 42)
+    {.emit: "console.assert(" & astToStr(assertion) & ");".}
```

Comment:
the error message sucks, it gives zero context nor line number.
instead start with this:
```nim
    var a = ""
    a.addQuoted(astToStr(assertion))
    {.emit: ["console.assert(", assertion, ",",  a.cstring, ");"].}
```
and modify it to also file/line/col (same as in assertions.nim using `toLocation`)

Reply:
But is not meant to be better or equal than normal assertions, 
is just for when you need faster performance AND assertions, otherwise use normal assertions.

Reply:
suppose you use this in a program, it'll give you 0 context as to where an jsAssert failure was triggered; and since it keeps going on failure, you'll just get:
```
assert failed
assert failed
assert failed
```

how would the suggestion result in slower performance?

Reply:
Done, kinda similar message to Nim assertions.